### DISCUSSION TITLE
Download plugin

### DISCUSSION THREAD

**Discussion by PH5HP** [rejetto/hfs#307] [General]
Hi Rejetto,

What happens if some files were zipped and downloaded in one zip with the counter download stats? It seems that  the files which were selected to be zipped and after beeing downloaded are not logged in the counter.yaml file. It would be accurate to do so. Is this possible? PSE take a look of my screenshot. 

If I am correct a ZIP file of 233Mb has been created in the Raimeter directory and downloaded after that, but no sign of anything in the counter.yaml file. It is logical not to count the created ZIP-file, but it is also logical to count the files within the created ZIP file.

I looked at the download plugin .js file but I still do not understand how it works. Just little parts I do understand. I wanted for example make little changes of how the counted downloads are displayed in the main layout, but I did not find the way yet.

![image](https://github.com/rejetto/hfs/assets/140949730/e29f533a-675a-4e5e-afcf-302dc0730c28)

Kind regards,
Henk - PH5HP - ftp.ph5hp.nl (when my computer is online)

**Comment by rejetto** [comment]
hey Henk,
the plugin at the moment is not considering zipped downloads.
I already thought this behavior to be inconsistent, but I'm happy to have a feedback about it.

When you click "zip", that is a virtual zip file. The zip file is never created on the disk, so it is not "created in the directory".
If we want to count the files in the zip, I guess I should wait for the zip to be fully downloaded, and only then count the files.

> I wanted for example make little changes of how the counted downloads are displayed in the main layout, but I did not find the way yet.

Go in details and I'll try to help you.

**Reply by PH5HP** [reply]
Hi Rejetto,

Thank you for beeing so patient with me.

I wonder if it can be done what you are describing about the virtual ZIP file and the files within it. It would be great if all files in it were analyzed and counted like the normal files which are downloaded.

I also would nice to beeing able to turn the ZIP feature on or off.

I am trying to understand how to write plugins for myself, but the lack of knowledge of the API of HFS makes it difficult. However I just discovered a few days ago that you are working on a new modern version of HFS. SO maybe a learning curve for me.

I just want to alter the appearance and maybe the place where the number of counts of a file download is placed. Maybe behind the size of the file for example. I looked in the source file of the download plugin but I fail to see where that is done.

Kind regards,
Henk - PH5HP - ftp.ph5hp.nl

**Reply by rejetto** [reply]
you would turn the zip feature off?
something that's already technically possible is to hide the zip button (easy) and to hide the "download" link for folders, which is zip too (slightly less easy).
If you want to proceed like this, I can help you.

plugins API is available at https://github.com/rejetto/hfs/blob/main/dev-plugins.md
for any question, open a new discussion.

> Maybe behind the size of the file for example.  I looked in the source file of the download plugin but I fail to see where that is done.

because the order is not defined. The API currently doesn't allow you to change that, and I'm not very sure it should.
Anyway, CSS makes it possible. Something like this:
```css
.entry-details { display: flex;  gap: 0.3em; }
.entry-size { order: -1 }
```

**Reply by PH5HP** [reply]
Hi Rejetto,

Maybe it is possible to limit the ZIP-file and a choise to turn off? Maybe turn it off as a choise regarding for every directory in the SHARED FILES section? So that you can turn it off per directory?

Some people try to download the entire content of my server which is several 100 of Gigabytes.

Kind regards,
Henk - PH5HP - ftp.ph5hp.nl
![image](https://github.com/rejetto/hfs/assets/140949730/abcf7eaa-fdbf-40e3-a733-883b01a6a8ed)

**Reply by rejetto** [reply]
I implied that it's not currently possible to turn the zip off, but only to hide the feature.
I can consider adding a per-folder option, yes.

**Comment by PH5HP** [comment]
Hi Rejetto,

Another thing. I was monitoring the MONITOR page in  the ADMIN of HFS and I saw 3 connections at the same timedownload the same file to combine it at the end to one file. Nothing wrong with that.

But I wonder how the Download Counter plugin counts that. Does it count as 1 or does it count as more than one, because I saw in the counters.yaml that is was already downloaded 4 times and I am almost sure that it was not downloaded before.

Any thoughts on that one?

Kind regards,
Henk - PH5HP - ftp.ph5hp.nl

**Comment by rejetto** [comment]
it is currently counting it multiple times, and that's not good, I think.
I shall change it.

**Reply by rejetto** [reply]
solved in next release

**Comment by rejetto** [comment]
files in zip will be counted from next release

**Comment by rejetto** [comment]
https://github.com/rejetto/hfs/releases/tag/v0.47.0-alpha6

you can click "check for updates" and then go for "alpha6"
